homeassistant/components/command_line/binary_sensor.py

```diff
@@ -50,17 +54,21 @@
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Command line Binary Sensor."""
 
-    setup_reload_service(hass, DOMAIN, PLATFORMS)
+    await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
+
+    binary_sensor_config = vol.Schema(
+        TEMPLATE_ENTITY_BASE_SCHEMA.schema, extra=vol.REMOVE_EXTRA
```

Comment:
What's happening here, why do we use the base schema and remove extra keys?
Also, why is the `PLATFORM_SCHEMA` not updated to extend the base schema, will that happen in follow-up PRs?

Reply:
`binary_sensor_config` is used for `TemplateEntity` so it's to align the config with what `TemplateEntity` expects and remove keys not used by it.
`PLATFORM_SCHEMA` isn't updated as it's intended to move to integration key instead in follow-up PR.

Reply:
Can remove `docs-missing` as `PLATFORM_SCHEMA` isn't expanded with the new options in this PR

Reply:
Thanks @gjohansson-ST, all clear 👍